sdk/src/main/java/com/bugsnag/android/Client.java

```diff
@@ -632,20 +629,23 @@ void setUserName(String name, boolean notify) {
     }
 
     @SuppressWarnings("ConstantConditions")
+    @Deprecated
     void setErrorReportApiClient(@NonNull ErrorReportApiClient errorReportApiClient) {
         if (errorReportApiClient == null) {
             throw new IllegalArgumentException("ErrorReportApiClient cannot be null.");
         }
-        this.errorReportApiClient = errorReportApiClient;
+        DeliveryCompat compat = (DeliveryCompat) config.getDelivery();
```

Comment:
It won't crash in the [current implementation](https://github.com/bugsnag/bugsnag-android/pull/311/files#diff-18f38731c79b15200b4f49b6eb495a3cR521) - but I agree we could use an `instanceof` check here.

Reply:
It would crash if someone called `setDelivery(new MyCustomDelivery())` and then `setErrorReportApiClient()`, as `getDelivery()` wouldn't have contained the property `errorReportApiClient`. The fix looks good though.